### DISCUSSION TITLE
Marp is an excellent tool, and it would be more excellent  while it supporting graphviz!

### DISCUSSION THREAD

**Discussion by k7** [marp-team/marp#219] [Ideas] [answered]
Yes，drawing languages such like graphviz and mermaid is very important for some creators.

**Comment by yhatt** [comment, marked as answer]
Yes, community has already enjoyed diagrams in Marp, powered by compatible markdown-it plugins ([GraphViz](https://www.npmjs.com/search?q=markdown-it%20graphviz&ranking=popularity), [mermaid](https://www.npmjs.com/search?q=markdown-it%20mermaid&ranking=popularity), and [PlantUML](https://www.npmjs.com/search?q=markdown-it%20plantuml&ranking=popularity)) or "Diagram images as a Service" such as [kroki.io](https://kroki.io/), [Gravizio](https://g.gravizo.com/), etc.

_Marp MUST be a minimalistic ecosystem to keep Markdown compatibility with everywhere._ If you want diagrams, plug the diagram support as you like.

> The plugin support is Marp's advanced part so it works on [Marp CLI](https://github.com/marp-team/marp-cli).

### Example: markdown-it-plantuml


```markdown
# UML example:

@startuml
actor actor
actor/ "actor/"
agent agent
artifact artifact
boundary boundary
card card
circle circle
cloud cloud
collections collections
component component
control control
database database
entity entity
file file
folder folder
frame frame
hexagon hexagon
interface interface
label label
node node
package package
person person
queue queue
rectangle rectangle
stack stack
storage storage
usecase usecase
usecase/ "usecase/"
@enduml
```

```javascript
// engine.js
module.exports = ({ marp }) =>
  marp.use(require('markdown-it-plantuml'), {
    // Wrap in <p> for better rendering in some built-in themes
    render: (...args) =>
      `<p>${marp.markdown.renderer.rules.image(...args)}</p>`,
  })
```

```bash
$ npm i --save @marp-team/marp-cli markdown-it-plantuml
$ npx marp --engine ./engine.js your-markdown.md
```

![plantuml](https://user-images.githubusercontent.com/3993388/143237060-c1269597-8739-48c4-bb4f-ae591cc359de.png)

### Related

- Mermaid
  - https://github.com/marp-team/marp/discussions/207
- PlantUML
  - https://github.com/marp-team/marp-vscode/issues/116
  - https://github.com/marp-team/marp-core/issues/112 (Fixed)
  - https://github.com/marp-team/marp-core/issues/193 (Fixed in the above example)
